src/main/java/com/locat/api/domain/user/service/impl/UserEndpointServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.locat.api.domain.user.service.impl;
+
+import com.locat.api.domain.user.entity.UserEndpoint;
+import com.locat.api.domain.user.service.UserEndpointService;
+import com.locat.api.infrastructure.repository.user.UserEndpointRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class UserEndpointServiceImpl implements UserEndpointService {
+
+    private final UserEndpointRepository userEndpointRepository;
+    @Override
+    public List<String> findUserEndpointArnsByUserId(Long userId) {
+        List<UserEndpoint> userEndpoints = this.userEndpointRepository.findByUserId(userId);
+
+        if (userEndpoints.isEmpty()) {
+            throw new RuntimeException("No user endpoint ARN found for userId=" + userId);
+        }
+
+        return userEndpoints.stream()
+                .map(UserEndpoint::getEndpointArn)
+                .toList();
+    }
```

Comment:
- 엔티티의 특정 필드만을 반환하는 것보다, 엔티티를 그대로 반환하는 게 확장성 측면에서 유리할 것 같아요.
    - 특정 필드만 반환하게 되면, 이후 다른 필드들이 필요할 때마다 메서드 시그니처를 변경하거나 새로운 메서드를 추가해야 해요
    - 조회된 데이터의 일부만 반환하면, 값이 호출자 쪽에서 다른 측면으로 해석되거나 본래 목적이 훼손될 수 있을 것 같아요
- 조회된 데이터가 없는 경우에도 예외를 던지기보다, 빈 List를 반환하는게 더 적절할 것 같아요.
    - 데이터 조회 후 처리를 호출자에게 위임해 더 다양한 시나리오(데이터 유무에 따라 다른 처리를 수행하거나, 기본값을 설정하거나, 메세지를 남기거나 등)에 유연하게 대응할 수 있을 것 같아요!

```suggestion
    public List<UserEndpoint> findUserEndpointByUserId(Long userId) {
        return this.userEndpointRepository.findByUserId(userId);
    }